gitlab/resource_gitlab_application.go

```diff
@@ -0,0 +1,137 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabApplicationCreate,
+		Read:   resourceGitlabApplicationRead,
+		Delete: resourceGitlabApplicationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"redirect_uri": {
```

Comment:
What do you think about changing the `Type` of `redirect_uri` and `scopes` to `schema.TypeList`, and handling the comma-joining for the user with `strings.Join(...`?

Reply:
Yep, I'm in favor of doing this. I'll take it from #366 as mentioned above.